### PR TITLE
Visitor can see articles in categories

### PR DESCRIPTION
**Pull request by johannaanden**
## Features
- Creates Category model
- Adds associations
- Creates factorybot
- Adds Unit test for category model
- Adds Acceptance test for category model
- Adds category nav bar as a partial
- Adds links to show the news associated to the selected categories

### PT Link
[Link](https://www.pivotaltracker.com/story/show/163144775)

### Screenshoot
screenshot 2019-01-11 at 12 51 07

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Category < ApplicationRecord
+  validates_presence_of :name
+  has_many :article
```
Comment (oliverochman) [inline app/models/category.rb]: change this to :articles

File: spec/models/category_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  describe 'DB table' do
+    it { is_expected.to have_db_column :name }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+  end
+
+  describe 'Associations' do
+    it { should have_many(:article) }
```
Comment (oliverochman) [inline spec/models/category_spec.rb]: here as well

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/models/category.rb (hunk 1 above)
Comment (GergKllai1) [reply app/models/category.rb]: Done

File: spec/models/category_spec.rb (hunk 2 above)
Comment (GergKllai1) [reply spec/models/category_spec.rb]: Done

File: features/Visitor_can_view_articles_in_categories.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            | Weather | 2  |
+        Given the following article exists:
+            | title   | description         | content                      | journalist | category_id |
+            | News    | This is a news      | I like to eat cheese!        | Jon        | 1           |
+            | Article | This is an article  | I do not like to eat cheese! | Greg       | 2           |
+        And I visit the landing page
+
+    Scenario: Visitor can see based on categories
+        When I click on 'Weather'
+        Then I should see "Article"
```
Comment (HannaSwDn) [inline features/Visitor_can_view_articles_in_categories.feature]: Do you want to use single or double quotes?

File: spec/models/article_spec.rb [hunk 4]
```diff
@@ -15,6 +15,10 @@
     it { is_expected.to validate_presence_of :journalist } 
   end
 
+  describe 'Associations' do
+    it { should belong_to(:category) }
+  end
+
   describe 'Factory' do
     it 'should have valid Factory' do
       expect(FactoryBot.create(:article)).to be_valid
```
Comment (HannaSwDn) [inline spec/models/article_spec.rb:24]: You don't have an end here

Comment (GergKllai1) [reply spec/models/article_spec.rb:24]: We have :)

File: features/Visitor_can_view_articles_in_categories.feature (hunk 3 above)
Comment (GergKllai1) [reply features/Visitor_can_view_articles_in_categories.feature]: We went single all the way

**Review by oliverochman — APPROVED** [review]

**Pull request merged**